unity-renderer/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/NFTShape/NFTInfoFetcher.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using DCL;
+using DCL.Helpers.NFT;
+using Newtonsoft.Json;
+using NFTShape_Internal;
+using UnityEngine;
+using UnityGLTF.Loader;
+
+public interface INFTInfoFetcher
+{
+    void FetchNFTImage(string address, string id, Action<NFTInfo> OnSuccess, Action OnFail);
+    void Dispose();
+}
+
+public class NFTInfoFetcher : INFTInfoFetcher
+{
+    internal Coroutine fetchCoroutine;
+
+    public void Dispose()
+    {
+        if (fetchCoroutine != null)
+            ;
+        CoroutineStarter.Stop(fetchCoroutine);
+        fetchCoroutine = null;
+    }
+
+    public void FetchNFTImage(string address, string id, Action<NFTInfo> OnSuccess, Action OnFail) { fetchCoroutine = CoroutineStarter.Start(FetchNFTImageCoroutin(address, id, OnSuccess, OnFail)); }
+
+    private IEnumerator FetchNFTImageCoroutin(string address, string id, Action<NFTInfo> OnSuccess, Action OnFail)
```

Comment:
```suggestion
    private IEnumerator FetchNFTImageCoroutine(string address, string id, Action<NFTInfo> OnSuccess, Action OnFail)
```
Please be more careful with the naming...